department-of-veterans-affairs/caseflow
# TYLERB/APPEALS-9185: Add an unrecognized claimant's POA (HLR)

Resolves [APPEALS-9185](https://vajira.max.gov/browse/APPEALS-9185) and [APPEALS-9186](https://vajira.max.gov/browse/APPEALS-9186). It also resolves [APPEALS-10768](https://vajira.max.gov/browse/APPEALS-10768)
### Description
Added the ability to add an unlisted claimant's POA during a higher-level review or supplemental claim intake. Added specific form field handling that differs between individual type POA and organization type POA by what fields are optional/required.

### Acceptance Criteria
- [x]  'Yes' is selected for the VA Form 21-22 question
  - [x] User is routed to “Add Claimant’s POA” screen
  - [x] User is prompted to search for the Representative’s name
- [x] Search for existing attorney
  - [x] User will select the name to populate the field
  - [x] User is able to see the Representative's address and the 'Continue to next step' button is enabled
- [x] Creating a new unlisted attorney 
  - [x] User can select Name not listed from the Representative's name field
    - [x] New unlisted organization attorney
      - [x] Required fields
        - [x] Organization Name
        - [x] Street Address 1
        - [x] City
        - [x] State
        - [x] Country (Default to “USA”)
        - [x] All other fields implied optional
    - [x] New unlisted individual attorney
      - [x] Required fields
        - [x] First name
        - [x] Last name
        - [x] All other fields implied optional
  - [x] Verify Modal Information 
    - [x] Claimant
      - [x] Claimant name
      - [x] Claimant Street address 1
      - [x] Claimant Street address 2
      - [x] Claimant Street address 3
      - [x] Claimant City, State, Zip, Country
    - [x] Claimant's PoA
      - [x] POA Name
      - [x] POA Street Address 1
      - [x] POA Street Address 2
      - [x] POA Street Address 3
      - [x] POA City, State, Zip, Country
- [x] Cancel Intake
  - [x] Confirm Cancel button has primary button css
  - [x] Confirm cancel is disabled until a radio selection has been made unless it is other. 
    - [x] If it is other, then the text field must not be blank.  

### Testing Plan
1. Possibly veteran data creation. If you are running it locally you might also have to enable feature flags
```
make enable-feature-flags
```
2. Swap to an intake user.  BVAISHAW (VACO) is a common one.
3. Click on the mail intake tab and click the start link or navigate to the local url /intake (e.g. http://localhost:3000/intake/)
4. Select Decision Review Request: Higher-Level Review — VA Form 20-0996 from the intake dropdown to start a high level review intake
5. Click the continue to search button
6. Enter a valid veteran ssn into the input field and click the search button or hit enter to advance
7. Fill out all the fields the review request page.
8. Select 'Yes' for the field Is the claimant someone other than the Veteran?
9. Select Claimant not listed for the field Please select the claimant listed on the form. If the claimant is not listed, please select "Claimant not listed" and add their information in the next step.
10. Click the Continue to the next step button
11. Select a relationship type and fill out the form using the indicated required fields.
12. Select the yes radio button for the Do you have a VA Form 21-22 for this claimant? form field
![image](https://user-images.githubusercontent.com/109369527/199067669-044ae755-e90d-4877-a8fb-1c33e983b6e4.png)

13. Verify that the continue to next step button is enabled and proceed to the add PoA Page.
14. Enter an attorney name into the Representative's name searchable dropdown field e.g. Billy Leannon
15. Select the attorney's name from the list of avaiable attorneys and verify the attorney's address is displayed on the page
![image](https://user-images.githubusercontent.com/109369527/199068756-b47fae18-ee55-4c9e-9109-45c31002626e.png)

16. Select the continue to next step button
17. Verify the selected attorney's information is preset in the confirmation modal along with the claimant information

18. Verify that the claimant type and information appears on add issues page and complete the intake
19. Verify the poa is attached to the claimant in the rails console.
```ruby
# Get the newly created review task and chain the method calls in order to get the poa partipant id that you selected
poa_id = DecisionReviewTask.last.appeal.claimant.unrecognized_appellant.poa_participant_id

# Use that Poa_id to get the attorney
attorney = BgsAttorney.where(participant_id: poa_id)

```
![image](https://user-images.githubusercontent.com/109369527/199078762-16315176-f882-4631-934d-768e2060df52.png)

20. Repeat steps 1-13
21. Enter "Name not listed" into the Representative's name searchable dropdown field.
![image](https://user-images.githubusercontent.com/109369527/199079527-11cbfe05-253e-48a3-84d8-2a336cfa944a.png)

22. Select the organization radio button from the Is the representative an organization or individual? form field
23. Verify the optional/required fields from the AC with the continue to next step button
24. Continue to next step in the review
25. Verify the newly created organization unrecognized attorney appears in the modal along with the claimant information
26. Verify that the claimant type and information appears on add issues page and complete the intake
27. Verify the newly created poa is attached to the claimant in the rails console.
```ruby
# Verify that the attributes contain the information that was entered in the form
new_attorney = DecisionReviewTask.last.appeal.claimant.unrecognized_appellant.unrecognized_power_of_attorney
```
28. Repeat steps 1-13.
29. Enter "Name not listed" into the Representative's name searchable dropdown field.
30. Select the individual radio button from the Is the representative an organization or individual? form field
31. Verify the optional/required fields from the AC with the continue to next step button
32. Continue to next step in the review
33. Verify the newly created individual unrecognized attorney appears in the modal along with the claimant information
34. Verify that the claimant type and information appears on add issues page and complete the intake
35. repeat step 27 to verify that the newly created poa is attached to the claimant in the rails console
36. Repeat all steps as a Supplemental Claim intake for 9186 as well. The AC is identical.
37. At some point during the intake cancel the intake and verify that the submit button is disabled until a reason has been selected or the other text box has been filled in.

### Storybook Story
*For Frontend (Presentational) Components*
- [ ] Updated the AddPoaPage Storybook file
